### DISCUSSION TITLE
RJ45 as VLAN

### DISCUSSION THREAD

**Discussion by awolff64** [coreemu/core#759] [Q&A] [answered]
My current understanding is that I connect my network application to my simulated network via RJ45 ports.
These RJ45 ports are related to existing interfaces on simulation.

I interface into the simulation by means of single Ethernet interface with multiple VLANs on it. To connect the VLAN I have to create the VLAN interfaces on the simulation host. These are then referenced in the RJ45 ports.

This is a bit static. I.e. I create/provision the interfaces on the simulation host. These are the interface I can refer in the simulation.

Is there a more dynamic way to do this?
E.g. a VLAN-RJ45 which takes a physical interface and a VLAN ID as configuration.
When running a simulation it automatically creates the VLAN interface.

**Comment by bharnden** [comment, marked as answer]
Assuming you have to run the software outside and use RJ45, it is a decent idea. Otherwise you could just run it within a normal node.

I do not believe there is any more built in dynamic way of doing this. You could drive your scenario from a python gRPC script or python directly and then automate whatever you want dynamically. That would be the current solution.

This use case isn't something I see being heavily used/requested, but something to consider. I appreciate the input.
